### DISCUSSION TITLE
Install rabbitmq server on a mapped network drive

### DISCUSSION THREAD

**Discussion by gustavo-ojeda** [rabbitmq/rabbitmq-server#12664] [Questions]
### Describe the bug

I am trying to install the latest versions of rabbitmq server on a mapped network drive using Windows (10 specifically). The Folder associated to the drive is in a different machine.

In order to do this I've tried setting HOMEDRIVE and HOMEPATH variables to the selected drive, let's say volume Z. 

### Reproduction steps

1. Set HOMEDRIVE to Z:
2. Set HOMEPATH to \
3. Start the installer (exe file) using Admin privileges and select the install folder in Z:\RabbitMQ. All the files are copied correctly. 
4. Select Start service option.

### Expected behavior

The service is started and work as expected. The installation is completed.

### Additional context

The actual behavior shows an error while starting the service. So after disabling the Start service option and closing the installer dialog I tried to start it manually from a Terminal (using Admin access) and I got this:

```
> rabbitmq-service.bat remove
> set HOMEDRIVE=Z:
> rabbitmq-service.bat install
> rabbitmq-service.bat start
The RabbitMQ service is starting.
The RabbitMQ service could not be started.

A system error has occurred.

System error 1067 has occurred.

The process terminated unexpectedly.
```

I've also tried using UNC paths but it doesn't work either.

**Comment by michaelklishin** [comment]
@gustavo-ojeda filing issues for questions in repositories that have Discussions enabled in late 2024 is just rude.

I am not at all a Windows expert but setting `HOMEDRIVE` in the shell may or may not be enough. Such "more permanent" things must be set system-wide and not in a specific shell section. IIRC both `HOMEDRIVE` and `HOMEPATH` must be set because the runtime's [shared secret path](https://www.rabbitmq.com/docs/cli#erlang-cookie) uses `%HOMEDRIVE%%HOMEPATH%`.

`rabbitmq-server.bat` and `rabbitmq-service.bat` are [both open source](https://github.com/rabbitmq/rabbitmq-server/tree/main/deps/rabbit/scripts). You are welcome to take a look at what they do and explicitly or implicitly expect.

And so is the [Windows installer code](https://github.com/rabbitmq/rabbitmq-packaging/), in case there might be something relevant there.

**Comment by lukebakken** [comment]
Don't do this - we don't test installation to mapped drives.

**Reply by michaelklishin** [reply]
And using network-attached storage for node's data directories is [not a good idea](https://www.rabbitmq.com/docs/production-checklist#storage) if other options exist.
